### PR TITLE
Migrate direct deposit pt2 - actions and reducers read only state

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

Continuing to build out the new direct deposit page to support a single form flow.

**COP Frontend specific changes**
- I added an exclude in the babel.config.json. This excludes the tests folder and mocks folder if they exist from being analyzed for unit test coverage, so that coverage generated by instanbul is more accurate. There might be other folders that should be included, but this is what authenticated experience needs, and is probably a good base.

**Auth Experience Changes**
- Actions and action names added. I tried to streamline the action names overall. Just fetch was added as an action for now, but I will be adding the update requests next... stay tuned you cool cats.
- Added the fetch action call to the root Profile application. This is guarded by access checks `togglesLoaded & shouldFetchDirectDeposit` so it is only called when the toggle is on for the new experience.
- ProfileSectionHeadline now accepts an optional dataTestId so that I could test the different headlines that contain the same text.
- the new DirectDeposit component got a little more content that is still mostly placeholder, and this will also be getting facelift next-ish. A quick and dirty useSelector was slapped in to show api data in plain text for now.
- Added mock data for the new `direct-deposits` endpoint. This mimics the existing CNP endpoint except it has the `isEduClaimAvailable` control information added as well.
- Reducer for new direct deposit state was created and add to the redux store.
  - I chose to make this state structure a bit flatter, and make directDeposit a root property of the state store, and also to simplify the property names for the state information itself.
  - Within the reducer logic I opted to refactor to using spread operators instead of the platform `set()` util as it is a bit confusing since it differs from lodash's set function and also isn't really needed to do the reducer stuff here so why not reduce the cruft...
  - Minor refactor of some of the naming and stuff in the DirectDepositClient class, but that class is kinda janky and was a poor decision for me making in the first place. I think I am going to straight remove it with the legacy code and opt for a more module / functional approach where possible. "seemed like a good ide at the time" 
  - Tests added 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/77466
^ I'm kinda just whittling away at this mini epic

## Testing done

- Unit tests added for reducers and actions
- E2E test for what little new DD page stuff was updated

## Screenshots

UI changes aren't really drastic or important at this part, since its still preliminary in that regard.

|         | Before | After |
| ------- | ------ | ----- |
| Desktop | Screenshot 2024-03-25 at 11 46 30 AM | Screenshot 2024-03-25 at 11 45 38 AM |

## What areas of the site does it impact?

Profile -> Direct Deposit (nothing shipped to production yet)

## Acceptance criteria

- [x] Main read-only fetch and state is set up. Unit tests cover the added code.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added

### CODE REVIEW

File: src/applications/personalization/profile/actions/directDeposit.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import { getData } from '~/platform/user/profile/utilities';
+
+import { API_STATUS } from '../constants';
+import { recordApiEvent } from '../util';
+
+// action types for direct deposit information
+
+// fetch actions
+export const DIRECT_DEPOSIT_FETCH_STARTED = 'DIRECT_DEPOSIT_FETCH_STARTED';
+export const DIRECT_DEPOSIT_FETCH_SUCCEEDED = 'DIRECT_DEPOSI_FETCH_SUCCEEDED';
```
Comment (allisonlu) [inline src/applications/personalization/profile/actions/directDeposit.js:12]: tiny typo here!

File: src/applications/personalization/profile/tests/actions/directDeposit.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    server.resetHandlers();
+    recordApiEventStub.restore();
+    captureErrorStub.restore();
+  });
+
+  after(() => {
+    server.close();
+  });
+
+  it('should dispatch DIRECT_DEPOSIT_INFORMATION_FETCH_SUCCEEDED with response on success', async () => {
```
Comment (allisonlu) [inline src/applications/personalization/profile/tests/actions/directDeposit.unit.spec.js:43]: non-blocking: would it make more sense to rename this to the actual action name `DIRECT_DEPOSIT_FETCH_SUCCEEDED`?

Comment (adamwhitlock1) [reply src/applications/personalization/profile/tests/actions/directDeposit.unit.spec.js:43]: oh snap, yea you caught one one that I missed. I have INFORMATION in the original action names and didn't like it, so I removed it and didn't remove it from the test description. Thanks!

File: src/applications/personalization/profile/tests/actions/directDeposit.unit.spec.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    });
+    expect(dispatchSpy.secondCall.args[0]).to.eql({
+      type: DIRECT_DEPOSIT_FETCH_SUCCEEDED,
+      response: base,
+    });
+
+    expect(recordApiEventStub.calledTwice).to.be.true;
+  });
+
+  it('should dispatch DIRECT_DEPOSIT_INFORMATION_FETCH_FAILED with response on error', async () => {
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/personalization/profile/tests/actions/directDeposit.unit.spec.js:69]: yep same as above will fix

**Review by allisonlu — DISMISSED** [review]
🎉  lgtm, the actions are being dispatched correctly and all unit tests are passing on local (waiting for e2e)

**Review by allisonlu — APPROVED** [review]

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**